packages/runner/src/storage/cache.ts

```diff
@@ -1965,14 +1973,22 @@ export class StorageManager implements IStorageManager {
 
   protected connect(space: MemorySpace): IStorageProviderWithReplica {
     const { id, address, as, settings } = this;
+
+    // Determine the correct spaceIdentity for this space
+    // For the workspace space (where spaceIdentity.did() === space), use spaceIdentity
+    // For other spaces (like home space), pass undefined
+    const spaceIdentityForSpace = this.spaceIdentity?.did() === space
```

Comment:
Ah, is this because we open multiple spaces here, and `spaceIdentity` set at the root session shouldn't affect other spaces? Will be good to remove the spaceIdentity throughout soon